defaultclient.go

```diff
@@ -242,3 +252,73 @@ func (client *DefaultClient) HealthCheck() bool {
 	}
 	return true
 }
+
+// ValidateAudienceScope validate audience and scope of user access token
+func (client *DefaultClient) ValidateAudienceScope(claims *JWTClaims, scope string) error {
+	if baseURI == "" {
+		err := client.getClientInformation(client.config, claims.Namespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	if scope == "" {
+		scope = defaultScope
+	}
+
+	if scope != claims.Scope {
+		return errors.New("scope isn't valid")
+	}
+
+	var isValid = false
+	for _, reqAud := range claims.Audience {
+		if reqAud == baseURI {
+			isValid = true
+		}
+	}
+
+	if !isValid {
+		return errors.New("audience isn't valid")
+	}
+
+	return nil
+}
+
+func (client *DefaultClient) getClientInformation(config *Config, namespace string) (err error) {
```

Comment:
this should be called in init too

Reply:
probably here: `client.StartLocalValidation()`

Reply:
we need a namespace from claims and StartLocalValidation doesnt have a namespace so I think its enough to run getClientInformation() if the baseURI is empty